### DISCUSSION TITLE
Best fitting datastructure to save poll/voting result of multiple items on algorand?

### DISCUSSION THREAD

**Discussion by alcyoneus-g** [reach-sh/reach-lang#1405] [Q&A] [answered]
We have several items(musics), each item needs to be voted yes/no.
Resembling the following structure:
```
[
  { musicId : 1,
    yesCount: 1,
    noCount: 2
  },
 ...
]
```
or
```
  { 
      music1 : {,
          yesCount: 1,
         noCount: 2
      },
     ...
}
```
We are working on the Algorand Network, and subsequently we have realized that we can't really use `Map` since it can be cleared by anyone?
So any suggestion/idea as to what would be a perfect fitting datastructure to use in our reach contract to model such system is much appreciated. thanks :)

**Comment by jeapostrophe** [comment, marked as answer]
A `Map` cannot be cleared by anyone. It can be cleared by the address of the key, because we compile them to local storage. In the next version of Algorand, we will convert `Map`s to use box storage which does not have this problem AND support arbitrary keys. If you don't plan to release soon, you can use that feature when it comes out.

It is possible to simulate box storage using local storage with special logic sig accounts, but this is very complex and we don't recommend it.

If you can't wait to use box storage, then basically you need to fit your data into the 8k storage, which means a limit to how many items can have votes. I think that lots of decentralized apps should be designed where each contract holds one thing... meaning you'd have one contract per music item with its own votes. These issues are discussed in the RSVP tutorial --- https://docs.reach.sh/tut/rsvp/#tut-rsvp
